### PR TITLE
Setup dockerfile and artifact text for artifact evaluation

### LINKED ISSUE #161: Make artefact for AE

### PR DESCRIPTION
**Pull request by bollu**
Closes #161

### CODE REVIEW

**Comment by bollu** [comment]
This needs to be coordinated with @alexkeizer , so the script that regenerates our alive examples can also be potentially run in the Dockerfile.

File: artifact-evaluation/artifact.tex [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+\subsection{Artifact check-list (meta-information)}
+
+% {\em Obligatory. Use just a few informal keywords in all fields applicable to your artifacts
+% and remove the rest. This information is needed to find appropriate reviewers and gradually 
+% unify artifact meta information in Digital Libraries.}
+
+{\small
+\begin{itemize}
+  \item {\bf Program: } The code repsitory for our framework along with the test suite.
```
Comment (alexkeizer) [inline artifact-evaluation/artifact.tex:46]:
```suggestion
  \item {\bf Program: } The code repository for our framework along with the test suite.
```

File: artifact-evaluation/artifact.tex [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{Description}
+
+\subsubsection{Hardware dependencies}
+
+None.
+
+\subsubsection{Software dependencies}
+
+Docker is necessary to run our artifact. the Docker image has dependences needed to compile our framework with Lean4.
```
Comment (alexkeizer) [inline artifact-evaluation/artifact.tex:69]:
```suggestion
Docker is necessary to run our artifact. The Docker image has all dependencies needed to compile our framework with Lean4.
```

File: artifact-evaluation/artifact.tex [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+Access the docker image \texttt{opencompl-ssa} from
+(\url{http://doi.org/TODO-TODO-TODO-DOI}), then run:
+
+\begin{minted}[fontsize=\scriptsize]{text}
+$ docker load -i opencompl-ssa.docker
+$ docker run -it siddudruid/opencompl-ssa
+# | This clears the build cache,
+# | fetches the maths library from the build cache,
+# v and builds our framework.
```
Comment (alexkeizer) [inline artifact-evaluation/artifact.tex:88]:
```suggestion
# | and builds our framework.
```

**Review by alexkeizer — COMMENTED** [review]
Some typos I found after a quick skim.

I would like to either merge this soon, or extract the Dockerfile and regenerated `AliveAutoGenerated.lean` file into a separate PR and merge that one quickly.

**Comment by alexkeizer** [comment]
I've sorried out the newly regenerated Alive examples that were broken for now, so the CI should be happy.

The broken examples seem to fall in two categories: shifts that now need some extra reasoning with the newly modeled potential for UB, and cases where previously `simp` used to see through a constant which is now hidden behind a `BitVec.ofInt`

**Pull request merged**